test/flint/fmpz_factor-test.jl

```diff
@@ -56,4 +58,29 @@
   @test unit(F_minus1) == -1
   @test unit(F_minus1_ZZ) == -1
 
+for T in [ Int8,  Int16,  Int32,  Int64,  UInt8,  UInt16,  UInt32,  UInt64 ]
+  @test_throws ArgumentError factor(T(0))
+  fac99 = factor(T(99));
+  @test  unit(fac99) == 1
+  @test  typeof(unit(fac99)) == T
+  @test  length(fac99) == 2
+  @test  issetequal([a  for (a,_) in fac99], [3, 11])
+  @test  issetequal([e  for (_,e) in fac99], [2, 1])
+  for (a,_) in fac99
+    @test  typeof(a) == T
+  end
+end
+
+# Some tests for overflow
+@test_throws  InexactError  factor(Int8, 257)
```

Comment:
I'd rather test the 1-argument version:
```suggestion
@test_throws  InexactError  factor(Int8(257))
```